hs_core/tests/api/native/test_create_new_version_resource.py

```diff
@@ -0,0 +1,198 @@
+
+        # test to make sure a new unique identifier has been created for the new versioned resource
+        self.assertIsNotNone(new_res_generic.short_id, msg='Unique identifier has not been created for new versioned resource.')
+        self.assertNotEqual(new_res_generic.short_id, self.res_generic.short_id)
+
+        # test to make sure the new versioned resource has no content files
+        self.assertEqual(new_res_generic.files.all().count(), 0)
+        # test to make sure the new versioned resource has the correct identifier
+        self.assertEqual(new_res_generic.metadata.identifiers.all().count(), 1, msg="Number of identifier elements not equal to 1.")
+        self.assertIn('hydroShareIdentifier', [id.name for id in new_res_generic.metadata.identifiers.all()],
+                      msg="hydroShareIdentifier name was not found for new versioned resource.")
+        id_url = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), new_res_generic.short_id)
+        self.assertIn(id_url, [id.url for id in new_res_generic.metadata.identifiers.all()],
+                      msg="Identifier url was not found for new versioned resource.")
+
+        # test to make sure the new versioned resource is linked with the original resource via isReplacedBy and isVersionOf metadata elements
+        self.assertGreater(new_res_generic.metadata.relations.all().count(), 0, msg="New versioned resource does has relation element.")
+        self.assertIn('isVersionOf', [rel.type for rel in new_res_generic.metadata.relations.all()],
+                      msg="No relation element of type 'isVersionOf' for new versioned resource")
+        version_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), self.res_generic.short_id)
+        self.assertIn(version_value, [rel.value for rel in new_res_generic.metadata.relations.all()],
+                      msg="The original resource identifier is not set as value for isVersionOf for new versioned resource.")
+        self.assertIn('isReplacedBy', [rel.type for rel in self.res_generic.metadata.relations.all()],
+                      msg="No relation element of type 'isReplacedBy' for the original resource")
+        version_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), new_res_generic.short_id)
+        self.assertIn(version_value, [rel.value for rel in self.res_generic.metadata.relations.all()],
+                      msg="The new versioned resource identifier is not set as value for isReplacedBy for original resource.")
+
+        # test isReplacedBy is removed after the new versioned resource is deleted
+        hydroshare.delete_resource(new_res_generic.short_id)
+        self.assertNotIn('isReplacedBy', [rel.type for rel in self.res_generic.metadata.relations.all()],
+                         msg="isReplacedBy is not removed from the original resource after its versioned resource is deleted")
+
+
+    def test_new_version_raster_resource(self):
```

Comment:
I think they shouldn't need that. This was supposed to be implemented in a neutral fashion and should be able to be tested as such. If you have need test it across different resource types, then I'd say there are failures in how those types (or more likely their metadata) behave in a consistent way.

Now, that said, this I state as a goal but practically speaking we don't have the test coverage to back it up, so you may be right.